example/shortest_path.cpp

```diff
@@ -0,0 +1,164 @@
+
+// Copyright (c) 2023 Jiri Nytra (jiri.nytra at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <flux.hpp>
+
+#include <cassert>
+#include <cstddef>
+#include <iostream>
+#include <iterator>
+#include <limits>
+#include <optional>
+#include <ostream>
+#include <random>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+using namespace std::string_literals;
+
+auto intersperse = [](flux::sequence auto seq, std::string sep) -> flux::generator<std::string>
+{
+    bool first = true;
+    for (auto c: seq) {
+        if (first) {
+            co_yield c;
+            first = false;
+        } else {
+            co_yield sep;
+            co_yield c;
+        }
+    }
+};
```

Comment:
Definitely. I can make `intersperse()` adaptor after set adaptors.

Reply:
I was thinking about an even more general version, `round_robin(seq1, seq2, .... seqN)` which yields the first element from each sequence in turn, followed by the second element from each sequence in turn, etc. This could be used in combination with `flux::repeat()` to do the equivalent of `intersperse()`.

But that would be a **very** tricky adaptor to write...

Reply:
It sounds good.